Sources/TuistDependencies/SwiftPackageManager/Utils/PackageInfoMapper.swift

```diff
@@ -218,55 +229,70 @@ public final class PackageInfoMapper: PackageInfoMapping {
                             idToPackage: idToPackage,
                             targetDependencyToFramework: targetDependencyToFramework
                         )
                     }
             }
 
-        var externalDependencies: [ProjectDescription.Platform: [String: [ProjectDescription.TargetDependency]]] = .init()
-
+        // Products to targets map i.e., [platform: [product_name : [targets_included_in_product]]]
+        var productsToTargetDependencies: [ProjectDescription.Platform: [String: [ProjectDescription.TargetDependency]]] = .init()
         for platform in platforms {
-            externalDependencies[ProjectDescription.Platform.from(graph: platform)] = try packageInfos
+            productsToTargetDependencies[ProjectDescription.Platform.from(graph: platform)] = try packageInfos
                 .reduce(into: [:]) { result, packageInfo in
                     try packageInfo.value.products.forEach { product in
                         result[product.name] = try product.targets.flatMap { target in
-                            try ResolvedDependency.fromTarget(
-                                name: target,
-                                targetDependencyToFramework: targetDependencyToFramework,
-                                condition: nil
-                            )
-                            .map {
-                                switch $0 {
-                                case let .xcframework(path, _):
-                                    return .xcframework(path: path)
-                                case let .target(name, _):
-                                    // When multiple platforms are supported, add the platform name as a suffix to the target
-                                    let targetName = platforms.count == 1 ? name : "\(name)_\(platform.rawValue)"
-                                    return .project(target: targetName, path: Path(packageToFolder[packageInfo.key]!.pathString))
-                                case .externalTarget:
-                                    throw PackageInfoMapperError.unknownProductTarget(
-                                        package: packageInfo.key,
-                                        product: product.name,
-                                        target: target
-                                    )
-                                }
-                            }
+                            try getTargetDependencies(for: target,
+                                                      product: product.name,
+                                                      targetDependencyToFramework: targetDependencyToFramework,
+                                                      packageName: packageInfo.key,
+                                                      packageToFolder: packageToFolder,
+                                                      platform: platform,
+                                                      platformCount: platforms.count)
```

Comment:
Putting these refactoring in the middle of a big change makes much harder to track what is changing and what is just a refactoring